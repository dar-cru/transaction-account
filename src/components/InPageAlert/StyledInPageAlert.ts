import styled from "styled-components";

export const StyledContainer = styled.div<{ type: "error" | "success" }>`
  display: flex;
  background-color: transparent;
  padding: 10px 20px;
  text-align: center;
  max-width: 80%;
  margin: 20px auto;
  color: ${(props) => (props.type === "error" ? "red" : "green")};
  border-radius: 8px;
  border: 1px solid ${(props) => (props.type === "error" ? "red" : "green")};
  font-size: 14px;
`;
