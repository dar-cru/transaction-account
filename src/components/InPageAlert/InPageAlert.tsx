import { FC, ReactNode } from "react";
import { StyledContainer } from "./StyledInPageAlert";

type InPageAlertProps = {
  type: "error" | "success";
  message: ReactNode;
};

const InPageAlert: FC<InPageAlertProps> = ({ type, message }) => {
  return <StyledContainer type={type}>{message}</StyledContainer>;
};

export default InPageAlert;
