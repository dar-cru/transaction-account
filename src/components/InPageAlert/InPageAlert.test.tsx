import React from "react";
import { render, RenderResult } from "@testing-library/react";
import InPageAlert from "./InPageAlert";


describe("<InPageAlert />", () => {
  test("renders an alert with error styling", () => {
    const tree: RenderResult = render(<InPageAlert type ="error" message="Oh no, there's an error"/>);

    const { container } = tree;
    expect(container).toMatchSnapshot();
  });

  test("renders an alert with success styling", () => {
    const tree: RenderResult = render(<InPageAlert type ="success" message="YAY! Success!"/>);

    const { container } = tree;
    expect(container).toMatchSnapshot();
  });
});
