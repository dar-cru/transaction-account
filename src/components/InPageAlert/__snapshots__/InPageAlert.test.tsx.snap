// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InPageAlert /> renders an alert with error styling 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  padding: 10px 20px;
  text-align: center;
  max-width: 80%;
  margin: 20px auto;
  color: red;
  border-radius: 8px;
  border: 1px solid red;
  font-size: 14px;
}

<div>
  <div
    class="c0"
    type="error"
  >
    Oh no, there's an error
  </div>
</div>
`;

exports[`<InPageAlert /> renders an alert with success styling 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
  padding: 10px 20px;
  text-align: center;
  max-width: 80%;
  margin: 20px auto;
  color: green;
  border-radius: 8px;
  border: 1px solid green;
  font-size: 14px;
}

<div>
  <div
    class="c0"
    type="success"
  >
    YAY! Success!
  </div>
</div>
`;
